### ISSUE TITLE
[0.7] Out of memory on when attempting to load share embed

### ISSUE THREAD

**Issue by aubymori** [Rehike/Rehike#247]
When attempting to load the embed tab on Share, it seems that nothing happens. However, if you inspect the network requests, there's errors.

**Fatal error**:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 36864 bytes) in **C:\Users\Aubrey\src\Rehike\models\Rehike\Debugger\MYtWalker.php** on line **254**
**Fatal error**:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 32768 bytes) in **C:\Users\Aubrey\src\Rehike\modules\Rehike\Debugger\ErrorWrapper.php** on line **1**

**Comment by aubymori** [comment]
@syndiate Come have a look at this. I was going to assign you, but it wouldn't let me...

**Comment by YukisCoffee** [comment]
Just to add, this is definitely a memory leak. Either an infinite loop happened that allocated memory or this is a self-referencing object, though I thought I fixed recursive object views in YtWalker.

**Comment by syndiate** [comment]
@aubymori Works completely fine for me.

![image](https://user-images.githubusercontent.com/67703687/218271095-f327d0f9-8cec-45c6-b7cf-21c2228d6e5f.png)

**Comment by aubymori** [comment]
That's odd.

**Comment by syndiate** [comment]
Can you provide any additional logs for this error?

**Comment by aubymori** [comment]
That's all it gives.

**Comment by syndiate** [comment]
What video did you try embed on? It probably doesn't matter but really anything would help.

**Comment by aubymori** [comment]
Can't remember the video I tried it on, and besides, it happens on any video.

**Comment by syndiate** [comment]
Why does it error out at the defineAttr function?

**Comment by randomperson189** [comment]
Interestingly only the embed input text shows for me and not the preview

**Comment by syndiate** [comment]
@randomperson189 Can you send a screenshot with the expanded menu?

**Comment by YukisCoffee** [comment]
I don't get an out-of-memory error, but embed box doesn't load at all for me and I get a standard 404 instead. Your implementation accounts for `/share_ajax`, but it's using `/sharing_services` for me.

![image](https://user-images.githubusercontent.com/99076687/219144981-ccd3b6cc-b262-438c-b44d-5667ad4a9caf.png)

**Comment by randomperson189** [comment]
> @randomperson189 Can you send a screenshot with the expanded menu?

Wait hold on it works now, I didn't notice there was another read more button to expand the menu

**Comment by aubymori** [comment]
Doesn't occur on 0.7 for me. Closing.

**Comment by aubymori** [comment]
CORRECTION: This DOES occur on 0.7 Does NOT occur on 0.5.

**Comment by YukisCoffee** [comment]
Still applicable as of 2023/09/23 lol

I guess it's not a priority issue, then again. This only will occur with the debugger enabled. It's certainly a recursive object serialization issue with the debugger, although it's never been looked into as to why this occurs specifically with this embed model.

**Comment by YukisCoffee** [comment]
Okay, so after nuking my PC with a malloc bomb, I managed to figure out what was going on. Simple logic error in the attribute parser for the debugger. As intended, a pointer to the parent was pushed to a stack and then the pointer of the child was compared against it, however, the child handler would pop as if it were handled normally (exiting from the parent).

In debugging, I managed to generate a 1.6 GB large dump file that gave me a lot of information. By the time it crashed (although being logged), it managed to spit out a fucking 37,577 character long path name:

![image](https://github.com/Rehike/Rehike/assets/99076687/91d9721d-669f-438e-b661-ee0fc279d907)

This will be fixed soon.
